Make `sessionIds` optional in CAIP-27 (#320)
ChainAgnostic/CAIPs: CAIPs/caip-27.md

## Simple Summary

CAIP-27 defines a JSON-RPC method for a wallet-connected application to invoke a

> wallet invoke an JSON-RPC method in a specified context defined by a valid

Reviewer: ```suggestion
JSON-RPC method on a wallet in a specified context defined by a valid
```

[scopeObject][CAIP-217] and, optionally, tagged with a [sessionId][CAIP-171] for
maintaining session continuity when applicable.

## Abstract

This proposal has the goal of defining a standard method for decentralized
applications to invoke JSON-RPC methods from user agents (such as cryptocurrency

> wallets) directed to a given, previously-authorized target chain (such as nodes

Reviewer: ```suggestion
wallets) directed to a given, previously-authorized target context (such as nodes
```

## Simple Summary

CAIP-27 defines a JSON-RPC method for a wallet-connected application to invoke a
wallet invoke an JSON-RPC method in a specified context defined by a valid
[scopeObject][CAIP-217] and, optionally, tagged with a [sessionId][CAIP-171] for
maintaining session continuity when applicable.

## Abstract

This proposal has the goal of defining a standard method for decentralized
applications to invoke JSON-RPC methods from user agents (such as cryptocurrency
wallets) directed to a given, previously-authorized target chain (such as nodes
of a specific blockchain or consensus community within a protocol). It requires
a valid [scopeObject][CAIP-217]. It MAY be tagged with a [sessionId][CAIP-171]
for interoperability and composability if the [CAIP-25][] session which

> authorizes the request is keyed by a [sessionId][CAIP-171] (see [CAIP-316][] for

Reviewer 2: nit. `if the caip-25 request which authorized the session...`

Author: https://github.com/ChainAgnostic/CAIPs/pull/320/commits/d16684712df6a12638a723cd92bca7716c391f23

### Definition

The JSON-RPC provider is able to invoke a single JSON-RPC request accompanied by

> a [CAIP-2][] compatible `chainId`, and optionally scoped by the

Author: 🤔 can CAIP-27 target entire namespaces (as in all chainIds authorized under that namespace)? I wasn't aware of this... What are the use cases here? You want to simultaneously invoke a method across multiple chains? 

It doesn't seem like my changes introduce a need for more clarity on this point?

> a [CAIP-2][] compatible `chainId`, and optionally scoped by the

Reviewer: tbh i am wondering if it even should anymore, or if `wallet:<namespace>` makes more sense as the target in every usecase that currently exists.  I'm currently working with a non-evm group and it seems like there, the `wallet:<namespace>` approach makes more sense.  

At the time we wrote this into CAIP-27 originally, I thought maybe in DAG, coordination-chain, or sharded VM architectures it might not always make sense to target a specific network, but I have yet to encounter a namespace that *actually* works this way (i.e. has namespace-wide execution), there always seem to be subnets or private networks that still need a distinct targeting/routing system, and whatever non-chain tangle/dag/DHT/thing is always just mainnet/chainId 1 in every context I've yet read documentation about...

> a [CAIP-2][] compatible `chainId`, and optionally scoped by the

Reviewer: we should probably add non-normative guidance warning people NOT to use namespace-wide scopes unless they know what they're doing OR constrain it normatively to CAIP-2 scopes only, in a separate PR.
